devito/profiler.py

```diff
@@ -39,50 +95,208 @@ def add_profiling(self, block, name, omp_flag=None):
                            "(double)(end_%(n)s.tv_sec-start_%(n)s.tv_sec)+" +
                            "(double)(end_%(n)s.tv_usec-start_%(n)s.tv_usec)" +
                            "/1000000") %
-                          {"sn": self.name, "n": name}),
+                          {"sn": self.t_name, "n": name}),
             ])
         ]
 
-        return init + block + end
+        return init + code + end
 
-    @property
-    def get_timings_class(self):
+    def get_oi_and_flops(self, name, code, size):
+        """Calculates the total operation intensity of the code provided.
+        If needed, lets the C code calculate it.
+
+        :param name: The name of the field to be populated
+        :param code: The code to be profiled.
+        """
+        if name not in self.oi:
+            self.oi[name] = 0
+        if name not in self.flops_defaults:
+            self.flops_defaults[name] = 0
+
+        loads = {"stores": 0}
+
+        for elem in code:
+            if isinstance(elem, Assign):
+                assign_flops = self._get_assign_flops(elem, loads)
+                self.oi[name] += assign_flops
+                self.flops_defaults[name] += assign_flops
+            elif isinstance(elem, For):
+                for_flops = self._get_for_flops(name, elem, loads)
+                self.oi[name] += for_flops
+            elif isinstance(elem, Block):
+                block_oi, block_flops = self._get_block_oi_and_flops(name, elem, loads)
+                self.oi[name] += block_oi
+                self.flops_defaults[name] += block_flops
```

Comment:
What about AugmentedAssignments, such as Incr, Decr (i.e., +=, -=) ... ?

Maybe we should also have a default ``else`` branch which prints a warning if hit ?